### PR TITLE
Merge 0.12 into master

### PR DESCRIPTION
**Pull request by squat**
cc @thanos-io/thanos-maintainers

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.12.1
+0.12.2
```
Comment (metalmatze) [inline VERSION:1]: IIRC we had something like `0.13.0-dev` on master last time I merged back. Not sure if we "enforce" this somehow?!

Comment (squat) [reply VERSION:1]: :/ hm we don't have a strict process for this. master currently says `0.12.1` so this should be good to merge now. We can make a follow up PR to bump master to 0.13.0-dev and we should maybe put this in the release docs. WDYT?

Comment (bwplotka) [reply VERSION:1]: yes please. 


We removed this because `promu` was handling this, but not anyone ):

Comment (bwplotka) [reply VERSION:1]: let's fix in separate PR with release doc fix :+1:

**Comment by bwplotka** [comment]
I did squash :facepalm: 

I guess we should revert and do normal merge? ):

**Pull request merged**